dislib/data/classes.py

```diff
@@ -78,6 +79,39 @@ def extend(self, subsets):
         self._sizes.extend([None] * len(subsets))
         self._reset_attributes()
 
+    def transpose(self, n_subsets=None):
+        """ Transposes the Dataset.
+
+        Parameters
+        ----------
+        n_subsets : int, optional (default=None)
+            Number of samples in each transposed subset. If none, defaults to
+            the original number of subsets
+        """
+
+        if not n_subsets:
+            n_subsets = len(self._subsets)
+
+        sizes = [_subset_size(s) for s in self._subsets]
+        subsets_t = []
+        for i in range(n_subsets):
+            #
+            subsets_i = [_get_split_i(s, i, n_subsets) for s in self._subsets]
+            # merge the
+            new_subset = _merge_split_subsets(self.sparse, *subsets_i)
+            subsets_t.append(new_subset)
+
+        sizes = compss_wait_on(sizes)
```

Comment:
Are you sure you need to synchronize here? I think subset_sizes does the synchronization for you.